### PR TITLE
Liveliness QoS compatibility with OpenSplice for lease duration zero

### PR DESCRIPTION
**Pull request by dennis-adlink**
OpenSplice uses liveliness QoS lease duration zero as default, which results in the entity immediately getting not-alive in Cyclone. So we consider lease duration zero as infinite lease duration when communicating with an OpenSplice entity.

### CODE REVIEW

**Comment by eboasson** [comment]
@dennis-adlink the way you have chosen to rewrite the lease_duration setting is entirely correct, but on consideration, I believe it is the wrong approach to the problem: basically, OpenSplice bugs should be dealt with in OpenSplice unless there is a pressing need to work around them in Cyclone.

As far as I have been able to determine, it really concerns only two writers used in an OpenSplice-specific protocol for dealing with transient data. Cyclone supports proxy writers with lease duration 0, so all the consequence is simply that those writers would considered not-alive by Cyclone, except for very brief periods when a sample arrives. Therefore, as far as I can tell, there is no problem in most cases, and probably no significant one even for anyone implementing that particular protocol.

In short: thanks, but no thanks 😄
